pymc3/backends/base.py

```diff
@@ -519,16 +518,18 @@ def get_sampler_stats(self, stat_name, burn=0, thin=1, combine=True,
         except TypeError:
             chains = [chains]
 
-        results = [self._straces[chain].get_sampler_stats(stat_name, None, burn, thin)
-                   for chain in chains]
+        results = [
+            self._straces[chain].get_sampler_stats(stat_name, None, burn, thin) for chain in chains
+        ]
         return _squeeze_cat(results, combine, squeeze)
 
     def _slice(self, slice):
         """Return a new MultiTrace object sliced according to `slice`."""
         new_traces = [trace._slice(slice) for trace in self._straces.values()]
         trace = MultiTrace(new_traces)
         idxs = slice.indices(len(self))
-        trace._report = self._report._slice(*idxs)
+        trace.report._chain_warnings = self.report._chain_warnings
+        trace.report._n_draws = len(trace)
```

Comment:
does this work if `save_warmup` is True?

Reply:
I think slicing is done only when `discard_tune_sample=False`, so I'd say yes 🤷‍♂️